test/unit/TablelandTables.ts

```diff
@@ -3,23 +3,56 @@ import chai from "chai";
 import chaiAsPromised from "chai-as-promised";
 import { BigNumber } from "ethers";
 import { ethers, upgrades } from "hardhat";
-import { TablelandTables } from "../../typechain-types";
+import {
+  TablelandTables,
+  TestReentrancyRunSQLLegacy,
+  TestReentrancyMutateOne,
+  TestReentrancyMutate,
+} from "../../typechain-types";
 
 chai.use(chaiAsPromised);
 const expect = chai.expect;
 
 describe("TablelandTables", function () {
   let accounts: SignerWithAddress[];
   let tables: TablelandTables;
+  let runSqlLegacyReentrantController: TestReentrancyRunSQLLegacy;
+  let mutateOneReentrantController: TestReentrancyMutateOne;
+  let mutateReentrantController: TestReentrancyMutate;
 
   beforeEach(async function () {
     accounts = await ethers.getSigners();
     const Factory = await ethers.getContractFactory("TablelandTables");
+
     tables = await (
       (await upgrades.deployProxy(Factory, ["https://foo.xyz/"], {
         kind: "uups",
       })) as TablelandTables
     ).deployed();
+
+    await tables.deployed();
+
+    // Deploy test controller contracts
+    runSqlLegacyReentrantController = (await (
+      await ethers.getContractFactory("TestReentrancyRunSQLLegacy")
+    ).deploy(tables.address)) as TestReentrancyRunSQLLegacy;
```

Comment:
The contract factory for `TestReentrancyRunSQLLegacy` is different than `tables.address`